functional-tests/fn_bridge_deposit_happy.py

```diff
@@ -25,133 +26,163 @@ def __init__(self, ctx: flexitest.InitContext):
-            try:
-                _ = btcrpc.gettransaction(txid)
-                print("Found expected tx in mempool")
-                tx_published = True
-                break
-            except Exception as e:
-                print(e)
-        assert tx_published, "Tx was not published"
-        # time to mature FIXME: is this 6 blocks?
-        btcrpc.proxy.generatetoaddress(6, seq_addr)
-        time.sleep(1)
+    def test_deposit(
+        self, ctx: flexitest.RunContext, address: str, el_address: str, new_address=True
+    ):
+        """
+        Test depositing funds into the bridge and verifying the corresponding increase in balance
+        on the Strata side.
+        """
+        rollup_deposit_amount = DEFAULT_ROLLUP_PARAMS["deposit_amount"]
+
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        reth = ctx.get_service("reth")
+
+        self.logger.debug(f"EL address: {el_address}")
 
-        new_balance_3 = int(rethrpc.eth_getBalance(f"0x{el_address_1}"), 16)
-        assert new_balance_3 > original_balance_3, "balance did not increase"
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        self.logger.debug(f"Address: {address}")
+
+        # Make sure that the el_address has zero balance
+        original_balance = int(rethrpc.eth_getBalance(f"0x{el_address}"), 16)
+        self.logger.debug(f"Balance before deposit (EL address): {original_balance}")
+        if new_address:
+            assert original_balance == 0, "balance is not zero"
+        else:
+            assert original_balance > 0, "balance is zero"
+
+        # Generate Plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Send DRT from Address 1 to EL Address 1
+        self.make_drt(ctx, el_address, bridge_pk)
+
+        # Make sure that the balance has increased
+        new_balance = int(rethrpc.eth_getBalance(f"0x{el_address}"), 16)
+        self.logger.debug(f"Balance after deposit (EL address): {new_balance}")
+        assert new_balance > original_balance, "balance did not increase"
+
+        # Make sure that the balance is 10 BTC in Strata "wei"
+        expected_balance = rollup_deposit_amount * (10**10) * (2 if not new_address else 1)
+        assert new_balance == expected_balance, "balance is not the default rollup_deposit_amount"
```

Comment:
And this assertion should only check the difference i.e., `new_balance - expected_balance == rollup_deposit_amount`.

Reply:
good suggestion, done in 0e5bbb07597354c168807d85323b0957a07d36cd